src/Csv.php

```diff
@@ -540,6 +540,41 @@ public function auto($file = null, $parse = true, $search_depth = null, $preferr
         return $this->delimiter;
     }
 
+    /**
+     * Get total number of rows in csv without parsing whole data.
+     *
+     * @return bool|int
+     */
+    public function getTotalRowCount(){
+        if (empty($this->file_data)){
+            return false;
+        }
+
+        $this->_detect_and_remove_sep_row_from_data($this->file_data);
+
+        $pattern = sprintf('/("[^%s]*")|[^%s]*/i',$this->enclosure, $this->enclosure);
+        preg_match_all($pattern,$this->file_data, $matches);
+
+        foreach ($matches[0] as $match){
+            if (empty($match) || !preg_match("/{$this->enclosure}/", $match)){
```

Comment:
I'm confused about the `preg_match` call: I thought only numbers go into curly brackets? To look for a simple string, `strpos` seems like the better function to me - especially if someone has crazy enclosure chars that would need escaping.

Reply:
Right. Done